lib/chef/cookbook/cookbook_version_loader.rb

```diff
@@ -212,6 +224,24 @@ def chefignore
         @chefignore ||= Chefignore.new(File.basename(cookbook_path))
       end
 
+      def load_all_unignored_files
+        # List all directories that are not dotdirs, then list everything in
+        # those, *including* dotfiles and dotdirs.
+        # Finally, list all the _files_ at the root, including dotfiles, and
+        # merge those in.
```

Comment:
yeah, IDK either.  obviously you can't undo it if you ever wanted.  but all the things i can think of (`.git`, `.svn`, `.kitchen`) are always awful to actually upload anywhere.

there is the use case of something like `files/homedirs/lamont/.ssh/` but that's covered.

and if we went ahead and uploaded dotdirs that weren't in chefignore i think we'd create a flag day where everyone had to update their chefignore files to ignore all the dotdirs they didn't have to care about before...  lots of "why the @#$@#$ would i ever want to upload .git???"

so i think this makes sense, and i think 'fixing' it would be a chef-13 breaking change kind of thing where we'd need really good reasons/issues to put everyone else through the trouble....